scipy/stats/_stats_py.py

```diff
@@ -7169,51 +7169,50 @@ def ks_1samp(x, cdf, args=(), alternative='two-sided', mode='auto'):
+    >>> stats.ks_1samp(stats.uniform.rvs(size=100, random_state=rng),
+    ...                stats.norm.cdf)
+    KstestResult(statistic=0.5001899973268688, pvalue=1.1616392184763533e-23)
 
-    Reject null hypothesis in favor of alternative hypothesis: less
+    Indeed, the p-value is lower than our threshold of 0.05, so we reject the
+    null hypothesis that the data are normally distributed.
 
-    >>> stats.ks_1samp(x, stats.norm.cdf, alternative='greater')
-    KstestResult(statistic=0.018749806388..., pvalue=0.920581859791...)
-
-    Reject null hypothesis in favor of alternative hypothesis: greater
+    When testing random variates from the standard normal distribution, we
+    expect the data to be consistent with the null hypothesis most of the time.
 
+    >>> x = stats.norm.rvs(size=100, random_state=rng)
     >>> stats.ks_1samp(x, stats.norm.cdf)
-    KstestResult(statistic=0.100203351482..., pvalue=0.250616879765...)
+    KstestResult(statistic=0.05345882212970396, pvalue=0.9227159037744717)
 
-    Don't reject null hypothesis in favor of alternative hypothesis: two-sided
+    As expected, the p-value of 0.92 is not below our threshold of 0.05, so
+    we cannot reject the null hypothesis.
 
-    *Testing t distributed random variables against normal distribution*
+    Suppose, however, that the location of the normally distributed random
+    variates is shifted toward larger values. In this case, the CDF of the
+    random variates tends to be *less* than the CDF of the standard normal.
+    Therefore, we would expect the null hypothesis to be rejected with
+    ``alternative='less'``:
 
-    With 100 degrees of freedom the t distribution looks close to the normal
-    distribution, and the K-S test does not reject the hypothesis that the
-    sample came from the normal distribution:
+    >>> x = x + 0.5
+    >>> stats.ks_1samp(x, stats.norm.cdf, alternative='less')
+    KstestResult(statistic=0.21707976606981766, pvalue=6.435039066115298e-05)
 
-    >>> stats.ks_1samp(stats.t.rvs(100,size=100, random_state=rng),
-    ...                stats.norm.cdf)
-    KstestResult(statistic=0.064273776544..., pvalue=0.778737758305...)
+    and with ``alternative='two-sided'``:
 
-    With 3 degrees of freedom the t distribution looks sufficiently different
-    from the normal distribution, that we can reject the hypothesis that the
-    sample came from the normal distribution at the 10% level:
+    >>> stats.ks_1samp(x, stats.norm.cdf, alternative='two-sided')
+    KstestResult(statistic=0.21707976606981766, pvalue=0.00012870078132230596)
 
-    >>> stats.ks_1samp(stats.t.rvs(3,size=100, random_state=rng),
-    ...                stats.norm.cdf)
-    KstestResult(statistic=0.128678487493..., pvalue=0.066569081515...)
+    but not with ``alternative='greater'``:
+
+    >>> stats.ks_1samp(x, stats.norm.cdf, alternative='greater')
+    KstestResult(statistic=0.003970612076116944, pvalue=0.9941219063026167)
```

Comment:
I feel like we are missing a conclusion after having presented these 3 possibilities.

Reply:
I agree. The natural conclusions were like "It is so." or "it is as we expect." So I omitted one.

 How about
* stating all three expectations before the code,
* "Indeed, " or "As expected, "
* all three code snippets together
?

Reply:
What is missing is something about 'two-sided' and 'greater'. It reads good until 'less' where we get why you try this parameter, but then it's not clear about the other 2 tests.

Reply:
If you change the order to what most people would do it makes more sense: First the two-sided alternative, then greater, and finally less. 

> and with alternative='two-sided':

We expect H0 to be rejected in favor of the alternative = "two-sided", which it does

> but not with alternative='greater':

But there should not be enough evidence to reject H0 in favor of the alternative = "greater"

Finally if we try with the alternative = "less": 
blabla
We can reject with confidence H0 in favor of the alternative = "less" as expected.

Reply:
(my apologies, I don't know how to format this so it can be more easily implemented)

Reply:
> (my apologies, I don't know how to format this so it can be more easily implemented)
@ThijsStockmans feel free to use the suggestion feature so we can take in your inputs. It's this button and you can click on preview to have a look at what you suggest.
<img width="70" alt="Screenshot 2022-02-24 at 14 45 24" src="https://user-images.githubusercontent.com/23188539/155535800-611b1832-3de4-45bc-8c65-e75137b794a3.png">

Reply:
After shifting the data, I think I'm just going to do `alternative='less'` and not the others. Users should have an alternative in mind before they run the test rather than trying all three to see what the p-values are, so let's do the same. This would be more concise, it covers the use of `alrernative`, and it gives an example of the (sometimes unintuitive for KS users) meaning of a one-sided alternative.